base/compiler/compiler.jl

```diff
@@ -19,6 +19,8 @@ eval(m, x) = Core.eval(m, x)
 include(x) = Core.include(Compiler, x)
 include(mod, x) = Core.include(mod, x)
 
+const _typeof_prefer_singleton = Core.typeof
```

Comment:
nitpick: maybe the name is a bit confusing now? how about something like `_typeof_captured_variable`?

Reply:
Yeah, I think that would make more sense.